.github/workflows/process-benchmark-results.yaml

```diff
@@ -6,11 +6,88 @@ on:
 
 jobs:
   PROCESS_RESULTS:
+    permissions:
+      contents: 'write'
+      id-token: 'write'
     runs-on: ubuntu-latest
     steps:
-      - name: parse request
+      - name: Check out repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Parse triggering request
         shell: bash
         run: |
           echo "asset: ${{ github.event.client_payload.asset }}"
-          echo "event: ${{ github.event }}"
-          echo "client_payload: ${{ github.event.client_payload }}"
+          echo "(upstream) run_id: ${{ github.event.client_payload.run_id }}"
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Authenticate to Google Cloud
+        id: auth
+        uses: google-github-actions/auth@v2.1.3
+        with:
+            project_id: ${{ secrets.GCP_PROJECT }}
+            workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}
+            service_account: ${{ secrets.GCP_GHA_SA }}
+
+      - name: Set up Google Cloud SDK
+        uses: google-github-actions/setup-gcloud@v2
+        with:
+            version: '>= 473.0.0'
+
+      - name: Download GHA-formatted JSON files
```

Comment:
we need to obfuscate the bucket and asset layout.

Reply:
Good catch; this should be resolved now.